### PR TITLE
Add task chat command

### LINKED ISSUE #798: Slayer task (!task)
a slayer task plugin could be usefull

### PR DESCRIPTION
**Pull request by Spedwards**
Closes #798
Supersedes/closes #3645

Adds the `!task` chat command.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerConfig.java [hunk 1]
```diff
@@ -110,6 +110,17 @@ default boolean weaknessPrompt()
 		return true;
 	}
 
+	@ConfigItem(
+		position = 8,
+		keyName = "taskCommand",
+		name = "Task Command",
+		description = "Configures whether the quest point command is enabled<br> !task"
```
Comment (forsco) [inline runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerConfig.java]: This description should probably be updated

Comment (Spedwards) [reply runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerConfig.java]: Lol, whoops.

**Comment by deathbeam** [comment]
you are missing http-service part completely, so this will not work

File: runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java [hunk 2]
```diff
@@ -685,9 +713,96 @@ private void removeCounter()
...
+		return true;
+	}
+
 	//Utils
 	private String capsString(String str)
 	{
 		return str.substring(0, 1).toUpperCase() + str.substring(1);
 	}
+
+	private static String sanitize(String str)
```
Comment (Digbigpig) [inline runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java]: Deathbeam had this moved within java\net\runelite\client\util\Text.java. I think that would be the best practice to allow for future chat commands to utilize it.

File: http-service/src/main/java/net/runelite/http/service/chat/ChatController.java [hunk 3]
```diff
@@ -100,4 +100,26 @@ public int getKc(@RequestParam String name)
 		}
 		return kc;
 	}
+
+	@PostMapping("/task")
+	public void submitTask(@RequestParam String name, @RequestParam String task)
```
Comment (Adam-) [inline http-service/src/main/java/net/runelite/http/service/chat/ChatController.java]: Id like for this to probably have name, task, assigned and/or kills left, and location.

File: http-service/src/main/java/net/runelite/http/service/chat/ChatService.java [hunk 4]
```diff
@@ -78,4 +78,22 @@ public void setQp(String name, int qp)
...
+			value = jedis.get("task." + name);
+		}
+		return value;
+	}
+
+	public void setTask(String name, String task)
+	{
+		try (Jedis jedis = jedisPool.getResource())
+		{
+			jedis.setex("task." + name, (int) EXPIRE.getSeconds(), task);
```
Comment (Adam-) [inline http-service/src/main/java/net/runelite/http/service/chat/ChatService.java]: This should be a redis hash for the above mentioned properties.

**Comment by Adam-** [comment]
This is going to require server and client side validation the task info doesn't contain anything funny, such as `<`, `>`, or whitespace other than maybe spaces. Also maybe a length check too.

File: http-service/src/main/java/net/runelite/http/service/chat/ChatController.java [hunk 5]
```diff
@@ -100,4 +101,26 @@ public int getKc(@RequestParam String name)
...
+
+	@GetMapping("/task")
+	public String getTask(@RequestParam String name)
+	{
+		Map<String, String> properties = chatService.getTask(name);
+		if (properties == null || properties.get("task") == null)
+		{
+			throw new NotFoundException();
+		}
+		return properties.get("task") + "." + properties.get("amount") + "." + properties.get("initialAmount") + "." + properties.get("location");
```
Comment (Spedwards) [inline http-service/src/main/java/net/runelite/http/service/chat/ChatController.java]: I doubt this is the right way to go about this, but I couldn't think of any way to return this information in the format of a `String`.

**Comment by Spedwards** [comment]
> This is going to require server and client side validation the task info doesn't contain anything funny, such as `<`, `>`, or whitespace other than maybe spaces. Also maybe a length check too.

I can't see how anything "funny" would end up in any of the info, but where exactly would you want this validation placed?

**Pull request merged**